### PR TITLE
Implement listing of projection errors

### PR DESCRIPTION
**Pull request by imsdu**
Related to #3979 

The changes in the view routes will come in a later PR

### CODE REVIEW

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphViews.scala [hunk 1]
```diff
@@ -293,23 +293,23 @@ final class BlazegraphViews(
     * Return the existing indexing views in a project in a finite stream
     */
   def currentIndexingViews(project: ProjectRef): ElemStream[IndexingViewDef] =
-    log.currentStates(Predicate.Project(project)).evalMapFilter { envelope =>
```
Comment (imsdu) [inline delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphViews.scala:296]: Predicate is renamed to Scope to match the vocabulary we already use in other places

File: delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/FragmentEncoder.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package ch.epfl.bluebrain.nexus.delta.sourcing
+
+import doobie.util.fragment.Fragment
+
+/**
+  * A type class that provides a conversion from a value of type `A` to a doobie [[Fragment]].
+  */
+trait FragmentEncoder[A] {
```
Comment (imsdu) [inline delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/FragmentEncoder.scala:8]: This case class alloew to convert a A to a doobie fragment in the same fashion as we do for circe encoders for example

File: delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/model/FailedElemLogRow.scala [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import doobie.postgres.implicits._
+import io.circe.Encoder
+import io.circe.generic.semiauto.deriveEncoder
+
+import java.time.Instant
+
+/**
+  * The row of the failed_elem_log table
+  */
+final case class FailedElemLogRow(
```
Comment (imsdu) [inline delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/model/FailedElemLogRow.scala:21]: Extracted from `ProjectionStore` to its own file

File: delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/offset/Offset.scala [hunk 4]
```diff
@@ -68,6 +61,11 @@ object Offset {
   implicit final val offsetGet: Get[Offset] = Get[Long].map(from)
   implicit final val offsetPut: Put[Offset] = Put[Long].contramap(_.value)
 
+  implicit val offsetFragmentEncoder: FragmentEncoder[Offset] = FragmentEncoder.instance {
```
Comment (imsdu) [inline delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/offset/Offset.scala:64]: Change the implemetation to get a fragment to use the type class

**Review by olivergrabinski — APPROVED** [review]

**Pull request merged**